backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java

```diff
@@ -275,8 +275,11 @@ public Response updateBehaviorGroup(@Context SecurityContext sec,
         String orgId = getOrgId(sec);
 
         if (request.displayName != null) {
-            BehaviorGroup behaviorGroup = new BehaviorGroup();
-            behaviorGroup.setId(id);
+            BehaviorGroup behaviorGroup = behaviorGroupRepository.get(id);
```

Comment:
The purpose of this fetch is to get the `bundleId` from the database - that one is needed now and is not provided in the request.

I guess we could also update the check to have something like:
```
if (behaviorGroup.getId() != null) {
   query += " AND bundle.id != (SELECT bundle_id FROM BehaviorGroup WHERE id = :behaviorGroupId)
} else {
   query += " AND bundle.id != :bundleId"
}
``` 

WDYT?

Reply:
Is not

Reply:
Sorry, I got confused between the BG and the bundle id 😅

Reply:
The managed bean could be a problem indeed. Since the `displayName` field is changed, the bean will be considered dirty and therefore be updated at the end of the JPA transaction, after the other intended update.

You could replace
```
public BehaviorGroup get(UUID behaviorGroupId) {
    return entityManager.find(BehaviorGroup.class, behaviorGroupId);
}
```
with
```
public UUID getBundleId(UUID behaviorGroupId) {
    return entityManager.createQuery("SELECT bundle.id FROM BehaviorGroup WHERE id = :id", UUID.class)
    [...]
}
```

Reply:
Done!